webapp/channels/src/components/channel_notifications_modal/reset_to_default_button/index.tsx

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useMemo} from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import type {ChannelMembership, ChannelNotifyProps} from '@mattermost/types/channels';
+import type {UserNotifyProps, UserProfile} from '@mattermost/types/users';
+
+import {DesktopSound, NotificationLevels} from 'utils/constants';
+import {notificationSoundKeys} from 'utils/notification_sounds';
+
+export enum SectionName {
+    Desktop = 'desktop',
+    Mobile = 'mobile',
+}
+
+const VALID_SECTION_NAMES = Object.values(SectionName);
+
+export interface Props {
+    sectionName: SectionName;
+    userNotifyProps: UserProfile['notify_props'];
+
+    /** The user's selected channel notify props which is not yet saved */
+    userSelectedChannelNotifyProps: ChannelMembership['notify_props'];
+    onClick: (channelNotifyPropsDefaultedToUserNotifyProps: ChannelMembership['notify_props'], sectionName: SectionName) => void;
+}
+
+export default function ResetToDefaultButton(props: Props) {
+    function handleOnClick() {
+        const channelNotifyPropsDefaultedToUserNotifyProps = resetChannelsNotificationToUsersDefault(props.userNotifyProps, props.sectionName);
+        props.onClick(channelNotifyPropsDefaultedToUserNotifyProps, props.sectionName);
+    }
+
+    const areDesktopNotificationsSameAsDefault = useMemo(() => {
```

Comment:
Perhaps instead of passing all of these values into the this component and having it manually check what has changed for both sections, we could instead have the modal pass in only the settings which are in the section and the user-default values for those settings?

This component could then just loop through each value in the object that's passed in to determine if the settings are non-default, and it could reset them to the default value by returning only a `Partial` object to the modal with the settings which need to be changed instead of returning the entire object. I think we might be able to avoid having this component know anything about how to handle the specific settings if we were to do that

Reply:
we are using all the values of notify_props for both channel and user here, so it that case i am fine leaving this as is.